File: src/frontend/src/types/assessment.ts
export interface CampaignCreate {
  name: string
  description?: string
  scope: string
  department_id?: string
  team_id?: string
  start_date: string
  end_date: string
}

export interface CampaignRead {
  id: string
  name: string
  description: string | null
  scope: string
  status: string
  department_id: string | null
  team_id: string | null
  start_date: string
  end_date: string
  created_at: string
  updated_at: string
}

export interface AssessmentCreate {
  campaign_id: string
  assessee_id: string
  assessor_type: string
}

export interface ScoreInput {
  competency_id: string
  score: number
  comment?: string
}

export interface AssessmentScoreSubmit {
  scores: ScoreInput[]
  is_draft?: boolean
}

export interface AssessmentScoreRead {
  competency_id: string
  score: number
  comment: string | null
  is_draft: boolean
}

export interface UserBrief {
  id: string
  first_name: string
  last_name: string
}

export interface AssessmentRead {
  id: string
  campaign_id: string
  assessee: UserBrief
  assessor: UserBrief
  assessor_type: string
  status: string
  scores: AssessmentScoreRead[]
  created_at: string
}

export interface CampaignProgressRead {
  campaign_id: string
  total_assessments: number
  completed_assessments: number
  pending_assessments: number
  completion_pct: number
}

export interface AggregatedScoreRead {
  user_id: string
  competency_id: string
  final_score: number
  self_score: number | null
  peer_score: number | null
  tl_score: number | null
  dh_score: number | null
}

export interface PeerSelectionSet {
  peer_ids: string[]
}

File: src/backend/app/schemas/assessment.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel

from app.models.enums import AssessmentStatus, AssessorType, CampaignScope, CampaignStatus


class ScoreInput(BaseModel):
    competency_id: uuid.UUID
    score: int
    comment: str | None = None


class AssessmentCreate(BaseModel):
    campaign_id: uuid.UUID
    assessee_id: uuid.UUID
    assessor_type: AssessorType


class AssessmentScoreSubmit(BaseModel):
    scores: list[ScoreInput]
    is_draft: bool = True


class AssessmentScoreRead(BaseModel):
    competency_id: uuid.UUID
    score: int
    comment: str | None
    is_draft: bool
    model_config = {"from_attributes": True}


class UserBrief(BaseModel):
    id: uuid.UUID
    first_name: str
    last_name: str
    model_config = {"from_attributes": True}


class AssessmentRead(BaseModel):
    id: uuid.UUID
    campaign_id: uuid.UUID
    assessee: UserBrief
    assessor: UserBrief
    assessor_type: AssessorType
    status: AssessmentStatus
    scores: list[AssessmentScoreRead] = []
    created_at: datetime
    model_config = {"from_attributes": True}


class CampaignCreate(BaseModel):
    name: str
    description: str | None = None
    scope: CampaignScope
    department_id: uuid.UUID | None = None
    team_id: uuid.UUID | None = None
    start_date: date
    end_date: date


class CampaignRead(BaseModel):
    id: uuid.UUID
    name: str
    description: str | None
    scope: CampaignScope
    status: CampaignStatus
    department_id: uuid.UUID | None
    team_id: uuid.UUID | None
    start_date: date
    end_date: date
    created_at: datetime
    updated_at: datetime
    model_config = {"from_attributes": True}


class CampaignWeightsUpdate(BaseModel):
    department_head_weight: float
    team_lead_weight: float
    self_weight: float
    peer_weight: float


class CampaignProgressRead(BaseModel):
    campaign_id: uuid.UUID
    total_assessments: int
    completed_assessments: int
    pending_assessments: int
    completion_pct: float


class AggregatedScoreRead(BaseModel):
    user_id: uuid.UUID
    competency_id: uuid.UUID
    final_score: float
    self_score: float | None
    peer_score: float | None
    tl_score: float | None
    dh_score: float | None
    model_config = {"from_attributes": True}

class PeerSetRequest(BaseModel):
    peer_ids: list[uuid.UUID]
